Assets/Mirror/Runtime/UNetwork.cs

```diff
@@ -120,15 +120,14 @@ public static byte[] PackMessage(ushort msgType, MessageBase msg)
         }
 
         // unpack message after receiving
-        public static bool UnpackMessage(byte[] message, out ushort msgType, out byte[] content)
+        // -> callers create a new NetworkReader from content, so we might as
+        //    well return it instead of returning the content array.
+        public static bool UnpackMessage(byte[] message, out ushort msgType, out NetworkReader contentReader)
         {
-            NetworkReader reader = new NetworkReader(message);
+            contentReader = new NetworkReader(message);
```

Comment:
I would prefer to receive the reader and not the byte[].

Reply:
why exactly?

Reply:
there are 2 reasons:
1) we make no assumption that this reader is at the beginning of the message.   Perhaps one day we want to batch messages,  we could have many messages in the same byte [] so we should have as little assumptions here as possible.
2) it is not clear the scope of the reader.  we created it in this method but we use it outside this method.  In general,  I prefer to create an object,  and pass it down to whomever needs it and not pass it up.  That makes it clear where you create it and where you stop using it.